sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveIdentifierClause.scala

```diff
@@ -37,7 +37,7 @@ class ResolveIdentifierClause(earlyBatches: Seq[RuleExecutor[LogicalPlan]#Batch]
   override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUpWithPruning(
     _.containsAnyPattern(UNRESOLVED_IDENTIFIER, UNRESOLVED_IDENTIFIER_WITH_CTE)) {
     case p: PlanWithUnresolvedIdentifier if p.identifierExpr.resolved =>
-      executor.execute(p.planBuilder.apply(evalIdentifierExpr(p.identifierExpr)))
+      executor.execute(p.planBuilder.apply(evalIdentifierExpr(p.identifierExpr), p.otherPlans))
```

Comment:
I think a problem is when to invoke the plan builder. Shall we wait for all `p.otherPlans` to be resolved first? Without a clear trigger condition, we may hit rule order issues.

Reply:
Yeah, that is true. That is why I tried to add a couple of tests with interleaving of identifiers and non-identifiers. I can pass one more time through AstBuilder, as I believe only InsertInto and one more LogicalPlans were changed to have otherPlans as parameters.